src/Umbraco.Web.UI.Client/src/packages/core/components/ref-item/ref-item.element.ts

```diff
@@ -22,52 +17,28 @@ export class UmbRefItemElement extends UmbElementMixin(UUIRefElement) {
 		this.addEventListener(UUIRefEvent.OPEN, () => this.dispatchEvent(new Event('click')));
 	}
 
-	override render() {
-		return html`
-			<button
-				type="button"
-				id="btn-item"
-				tabindex="0"
-				@click=${this.handleOpenClick}
-				@keydown=${this.handleOpenKeydown}
-				?disabled=${this.disabled}>
-				${when(this.icon, () => html`<span id="icon"><umb-icon name=${this.icon ?? ''}></umb-icon></span>`)}
-				<div id="info">
-					<div id="name">${this.name}</div>
-					<small id="detail">${this.detail}</small>
-				</div>
-			</button>
-			<div id="select-border"></div>
-			<slot></slot>
-		`;
+	protected override firstUpdated(_changedProperties: PropertyValues): void {
+		super.firstUpdated(_changedProperties);
+
+		if (this.icon) {
+			const umbIcon = document.createElement('umb-icon');
+			umbIcon.setAttribute('name', this.icon);
+			this.shadowRoot?.querySelector('#icon')?.replaceWith(umbIcon);
```

Comment:
I think this is a little dangerous,, cause what if something caused Lit to re-render.. then it would not be overriden any more..

Could you try to just add it as a direct child, not in the shadowRoot and then set `slot='icon'`?

Reply:
It felt a bit hacky when I was developing that part. I did try a few other approaches, but I couldn't get the icon to display, the fallback icon kept being rendered.

For reference: https://github.com/umbraco/Umbraco.UI/blob/v1.12.2/packages/uui-ref-node/lib/uui-ref-node.element.ts#L109-L112

When you say "add it as a direct child", do you mean that it overrides the `render()` method? If so, I'm not quite sure how that would work.

I'm wondering if [`UUIRefNodeElement`](https://github.com/umbraco/Umbraco.UI/blob/v1.12.2/packages/uui-ref-node/lib/uui-ref-node.element.ts) could have a `protected renderIcon()` method added to it, that may simplify this. 🤔